### PR TITLE
Drop support for legacy platforms

### LINKED ISSUE #3052: Drop support for SQL Server older than 2008
The reasons are:
1. SQL Server 2008 is [the oldest officially supported](https://en.wikipedia.org/wiki/Microsoft_SQL_Server#Currently) by Microsoft version.
2. Same is about [AppVeyor](https://www.appveyor.com/docs/services-databases/).
3. Fixing some issues on the older versions (e.g. #3020) requires additional effort if at all possible.

### LINKED ISSUE #3053: Drop support for PostgreSQL <9.3
Anything below 9.3 is already EOL'd and not supported.
9.3 will be EOL'd on 9/2018 so depending on timeline it could be dropped too.
Keeping these for 3.0 would mean lock-in for another X years.

https://www.postgresql.org/support/versioning/

### LINKED ISSUE #3054: Drop support for SQL Anywhere <=12
SQL Anywhere 12.x and older are already EOL'd and not supported.

https://sqlanywhere-forum.sap.com/questions/26068/sql-anywhere-end-of-life-overview

### LINKED ISSUE #3058: Fate of PDOIbm driver
While working on adding DB2 to our CI stage, I found there are two drivers for DB2: IBMDB2 (ibm_db2 extension) and PDOIbm (pdo_ibm extension).
While the former works and seems to be supported, the latter is not supported by DriverManager, not documented and seems to be completely unused.

Decision should be made about this lonely driver.

### PR DESCRIPTION
**Pull request by Majkl578**
Drops:
 * PostgreSQL <9.3
 * SQL Server < 2008
 * SQL Anywhere < 16

closes #3052
closes #3053
closes #3054 
closes #3058

### CODE REVIEW

**Comment by Majkl578** [comment]
Changes:
* unsupported PDO DB2 driver dropped
* deprecated MsSQLKeywords dropped
* type mappings and keywords reordered

File: lib/Doctrine/DBAL/Platforms/SQLServerPlatform.php [hunk 1]
```diff
@@ -1403,6 +1410,10 @@ protected function initializeDoctrineTypeMappings()
             'varbinary' => 'binary',
             'image' => 'blob',
             'uniqueidentifier' => 'guid',
+            'datetime2' => 'datetime',
+            'date' => 'date',
+            'time' => 'time',
+            'datetimeoffset' => 'datetimetz',
```
[the remark this answers is not in the document]
Comment (Majkl578) [reply lib/Doctrine/DBAL/Platforms/SQLServerPlatform.php]: Done also for other platforms & keywords (separate commit).

File: lib/Doctrine/DBAL/Platforms/Keywords/OracleKeywords.php [hunk 2]
```diff
@@ -27,118 +27,118 @@ protected function getKeywords()
...
-            'GRANT',
-            'NUMBER',
-            'SQLBUF',
             'ASC',
-            'GROUP',
-            'OF',
-            'SUCCESSFUL',
             'AUDIT',
-            'HAVING',
-            'OFFLINE ',
```
Comment (Majkl578) [inline lib/Doctrine/DBAL/Platforms/Keywords/OracleKeywords.php]: Trailing space was a bug? Removed.

File: lib/Doctrine/DBAL/Platforms/Keywords/OracleKeywords.php [hunk 3]
```diff
@@ -27,118 +27,118 @@ protected function getKeywords()
...
-            'EXISTS',
-            'NOCOMPRESS',
-            'SESSION',
             'ALTER',
-            'FILE',
-            'NOT',
-            'SET',
             'AND',
-            'FLOAT',
-            'NOTFOUND ',
```
Comment (Majkl578) [inline lib/Doctrine/DBAL/Platforms/Keywords/OracleKeywords.php]: Trailing space was a bug? Removed.

File: lib/Doctrine/DBAL/Platforms/SQLAnywherePlatform.php [hunk 4]
```diff
@@ -1501,45 +1501,45 @@ protected function getVarcharTypeDeclarationSQLSnippet($length, $fixed)
...
-            'bit' => 'boolean',
-            'decimal' => 'decimal',
-            'double' => 'float',
-            'float' => 'float',
-            'int' => 'integer',
-            'integer' => 'integer',
-            'unsigned int' => 'integer',
-            'numeric' => 'decimal',
-            'smallint' => 'smallint',
-            'unsigned smallint', 'smallint',
```
Comment (Majkl578) [inline lib/Doctrine/DBAL/Platforms/SQLAnywherePlatform.php]: This was a bug - not a key-value mapping.

File: lib/Doctrine/DBAL/Platforms/SQLAnywherePlatform.php [hunk 5]
```diff
@@ -1501,45 +1501,45 @@ protected function getVarcharTypeDeclarationSQLSnippet($length, $fixed)
...
-            'tinyint' => 'smallint',
-            'unsigned tinyint', 'smallint',
```
Comment (Majkl578) [inline lib/Doctrine/DBAL/Platforms/SQLAnywherePlatform.php]: This was a bug - not a key-value mapping.

**Review by Majkl578 — COMMENTED** [review]
Highlighted 4 bugs that may be worth fixing in 2.x too.

**Review by morozov — APPROVED** [review]

**Review by Ocramius — APPROVED** [review]
:+1: 

Noticed that we don't have any explicit mapping for PostgreSQL > 9.4, but that's eventually for a separate patch, and probably not relevant for newer setups (10.x)

**Pull request merged**